src/libcrun/linux.c

```diff
@@ -3442,6 +3442,17 @@ validate_sysctl (const char *original_key, const char *original_value, const cha
   return crun_make_error (err, 0, "the sysctl `%s` requires a new %s namespace", original_key, namespace);
 }
 
+/* Best-effort attempt to give a better explanation why setting a sysctl could have failed. */
+static char *
+sysctl_error_reason (const runtime_spec_schema_config_schema *def, const char *name, const char *value, int namespaces_created, int errno_)
+{
+
+  if (strcmp (name, "net.ipv4.ping_group_range") == 0 && (errno_ == EINVAL) && (namespaces_created & CLONE_NEWUSER))
+    return xstrdup ("are all the group IDs mapped in the user namespace?");
```

Comment:
I think this error occurs when UID mappings are missing, hence it should say "are all the user IDs mapped" instead of "are all the group IDs mapped". In #1648 missing GID mappings throw another error